optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/custom/CustomShadowVariableDescriptor.java

```diff
@@ -134,7 +134,7 @@ private void linkShadowSources(DescriptorPolicy descriptorPolicy) {
                         + ") with refVariableName (" + refVariableName
                         + ") which is not a valid planning variable on entityClass ("
                         + refEntityDescriptor.getEntityClass() + ").\n"
-                        + entityDescriptor.buildInvalidVariableNameExceptionMessage(refVariableName));
+                        + refEntityDescriptor.buildInvalidVariableNameExceptionMessage(refVariableName));
```

Comment:
Wrong descriptor was used. The referenced variable is expected at the referenced entity descriptor, so `refEntityDescriptor` must be used to build the error message. `entityDescriptor` is the shadow planning entity that has the custom shadow variable which references a genuine variable by `refVariableName`.